### PR TITLE
Added charts for Pod DNS Chaos

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

- This PR adds the charts for Pod DNS chaos

### CODE REVIEW

File: charts/generic/pod-dns-chaos/pod-dns-chaos.chartserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - "kubernetes" 
+    - "generic" 
+    - "dns"
+  maturity: alpha 
+  minKubeVersion: 1.12.0 
+  provider: 
+    name: ChaosNative
+  maintainers: 
+    - name: gdsoumya 
+      email: gdsoumya@chaosnative.io
```
Comment (ishangupta-ds) [inline charts/generic/pod-dns-chaos/pod-dns-chaos.chartserviceversion.yaml]: .com

Comment (ishangupta-ds) [reply charts/generic/pod-dns-chaos/pod-dns-chaos.chartserviceversion.yaml]: cc: @gdsoumya

File: charts/generic/pod-dns-chaos/engine.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+kind: ChaosEngine
+metadata:
+  name: nginx-chaos
+spec:
+  appinfo:
+    appns: 'default'
+    applabel: 'app=nginx'
+    appkind: 'deployment'
+  # It can be true/false
+  annotationCheck: 'true'
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/engine.yaml]:
```suggestion
  annotationCheck: 'false'
```

File: charts/generic/pod-dns-chaos/engine.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  # It can be active/stop
+  engineState: 'active'
+  #ex. values: ns1:name=percona,ns2:run=nginx
+  auxiliaryAppInfo: ''
+  chaosServiceAccount: pod-dns-chaos-sa
+  monitoring: false
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/engine.yaml]: remove this

File: charts/generic/pod-dns-chaos/experiment.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      - name: TARGET_PODS
+        value: ''
+
+      # provide the name of container runtime, it supports docker, containerd, crio
+      - name: CONTAINER_RUNTIME
+        value: 'docker'
+
+      # provide the destination hosts
+      # chaos injection will be triggered for these destination hosts
+      - name: DESTINATION_HOSTS
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/experiment.yaml]: we can remove this

**Review by ispeakc0de — DISMISSED** [review]
Looks Good! We can test this once on MyHUB

**Review by ishangupta-ds — DISMISSED** [review]

**Review by ksatchit — DISMISSED** [review]

File: charts/generic/pod-dns-chaos/pod-dns-chaos.chartserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Source Code
+      url: https://github.com/litmuschaos/litmus-go/tree/master/experiments/generic/pod-dns-chaos
+    - name: Documentation
+      url: https://docs.litmuschaos.io/docs/getstarted/
+    - name: Video
+      url: 
+  icon:
+    - base64data: ""
+      mediatype: ""
+  chaosexpcrdlink: https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/pod-dns-chaos/experiment.ya
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/pod-dns-chaos.chartserviceversion.yaml]:
```suggestion
  chaosexpcrdlink: https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/pod-dns-chaos/experiment.yaml
```

File: charts/generic/pod-dns-chaos/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-dns-chaos-sa
+  namespace: default
+  labels:
+    name: pod-dns-chaos-sa
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/rbac.yaml:8]:
```suggestion
    name: pod-dns-chaos-sa
    app.kubernetes.io/part-of: litmus
```

File: charts/generic/pod-dns-chaos/rbac.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: pod-dns-chaos-sa
+  namespace: default
+  labels:
+    name: pod-dns-chaos-sa
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/rbac.yaml:16]:
```suggestion
    name: pod-dns-chaos-sa
    app.kubernetes.io/part-of: litmus
```

File: charts/generic/pod-dns-chaos/rbac.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    resources: ["chaosengines","chaosexperiments","chaosresults"]
+    verbs: ["create","list","get","patch","update"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: pod-dns-chaos-sa
+  namespace: default
+  labels:
+    name: pod-dns-chaos-sa
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/rbac.yaml:46]:
```suggestion
    name: pod-dns-chaos-sa
    app.kubernetes.io/part-of: litmus
```

File: charts/generic/pod-dns-chaos/experiment.yaml [hunk 9]
```diff
@@ -0,0 +1,102 @@
+apiVersion: litmuschaos.io/v1alpha1
+description:
+  message: |
+    injects dns failure in target pod container
+kind: ChaosExperiment
+metadata:
+  name: pod-dns-chaos
```
Comment (ispeakc0de) [inline charts/generic/pod-dns-chaos/experiment.yaml:7]:
```suggestion
  name: pod-dns-chaos
  labels:
    name: pod-dns-chaos
    app.kubernetes.io/part-of: litmus
    app.kubernetes.io/component: chaosexperiment
    app.kubernetes.io/version: latest
```

**Review by ispeakc0de — APPROVED** [review]

**Review by ishangupta-ds — APPROVED** [review]

**Pull request merged**